src/transformers/models/levit/modeling_levit.py

```diff
@@ -0,0 +1,727 @@
+
+        self.qkv = MLPLayerWithBN(embed_dim, self.out_dim_qkv)
+        self.activation = nn.Hardswish()
+        self.projection = MLPLayerWithBN(self.out_dim_proj, embed_dim, bn_weight_init=0)
+
+        points = list(itertools.product(range(resolution), range(resolution)))
+        len_points = len(points)
+        attention_offsets, idxs = {}, []
+        for p1 in points:
+            for p2 in points:
+                offset = (abs(p1[0] - p2[0]), abs(p1[1] - p2[1]))
+                if offset not in attention_offsets:
+                    attention_offsets[offset] = len(attention_offsets)
+                idxs.append(attention_offsets[offset])
+
+        self.attention_bias_cache = {}
+        self.attention_biases = torch.nn.Parameter(torch.zeros(num_heads, len(attention_offsets)))
+        self.register_buffer("attention_bias_idxs", torch.LongTensor(idxs).view(len_points, len_points))
+
+    @torch.no_grad()
+    def train(self, mode=True):
+        super().train(mode)
+        if mode and self.attention_bias_cache:
+            self.attention_bias_cache = {}  # clear ab cache
+
+    def get_attention_biases(self, device):
+        if self.training:
+            return self.attention_biases[:, self.attention_bias_idxs]
+        else:
+            device_key = str(device)
+            if device_key not in self.attention_bias_cache:
+                self.attention_bias_cache[device_key] = self.attention_biases[:, self.attention_bias_idxs]
+            return self.attention_bias_cache[device_key]
+
+    def forward(self, hidden_state):
+        batch_size, seq_length, _ = hidden_state.shape
+        qkv = self.qkv(hidden_state)
+        query, key, value = qkv.view(batch_size, seq_length, self.num_heads, -1).split(
+            [self.key_dim, self.key_dim, self.attention_ratio * self.key_dim], dim=3
+        )
+        query = query.permute(0, 2, 1, 3)
+        key = key.permute(0, 2, 1, 3)
+        value = value.permute(0, 2, 1, 3)
+
+        attention = query @ key.transpose(-2, -1) * self.scale + self.get_attention_biases(hidden_state.device)
+        attention = attention.softmax(dim=-1)
+        hidden_state = (attention @ value).transpose(1, 2).reshape(batch_size, seq_length, self.out_dim_proj)
+        hidden_state = self.projection(self.activation(hidden_state))
+        return hidden_state
+
+
+class LevitAttentionSubsample(nn.Module):
+    def __init__(self, input_dim, output_dim, key_dim, num_heads, attention_ratio, stride, resolution, resolution_):
+        super().__init__()
+        self.num_heads = num_heads
+        self.scale = key_dim**-0.5
+        self.key_dim = key_dim
+        self.attention_ratio = attention_ratio
+        self.out_dim_kv = attention_ratio * key_dim * num_heads + key_dim * num_heads
+        self.out_dim_proj = attention_ratio * key_dim * num_heads
+        self.resolution_ = resolution_
+
+        self.kv = MLPLayerWithBN(input_dim, self.out_dim_kv)
```

Comment:
```suggestion
        self.keys_values = MLPLayerWithBN(input_dim, self.out_dim_kv)
```
Sorry, but let's make it as clean as possible :)